picoquic/util.c

```diff
@@ -366,6 +366,26 @@ void picoquic_set64_connection_id(picoquic_connection_id_t * cnx_id, uint64_t va
     cnx_id->id_len = 8;
 }
 
+uint64_t picoquic_hash_addr(const struct sockaddr* addr)
+{
+    uint64_t h;
+
+    if (addr->sa_family == AF_INET) {
+        struct sockaddr_in* a4 = (struct sockaddr_in*)addr;
+        h = picohash_bytes((uint8_t*)&a4->sin_addr , 4);
+        h += 128ull * a4->sin_port;
+        h ^= 0x04040404;
```

Comment:
Why is that and the 0x06060606 below necessary?

Reply:
The idea was to add a different salt to the hash for IPv6 and IPv4 addresses. But yes, that probably not too useful.